engines/extpoints/interfaces.go

```diff
@@ -32,4 +32,4 @@ type EngineOptions struct {
 // Any error here will be fatal and likely cause the worker to stop working.
 // If an implementor can determine that the platform isn't supported at
 // compile-time it is recommended to not register the implementation.
-type EngineProvider func(options EngineOptions) (engines.Engine, error)
+type EngineProvider func(options *EngineOptions) (engines.Engine, error)
```

Comment:
I propose we make this a non-pointer type. See Note in the definition of `EngineOptions`.

Basically, I think when we have `<...>Options` where the struct is only used as input parameters for a method, then it makes sense to pass it by value.
It when we do struct grouping just because grouping the input parameters into a struct allows us to add new parameters without breaking old code.

I see this `<...>Options` pattern as passing key-word arguments in python. Ie. you'll always call `EngineProvider` as:
```go
engine, err := engineProvider(EngineOptions{
  Environment: ...
  ....
})
```

So it's just a pattern used to pass arguments, not an actual struct that we recommend people pass around as a value. Not a struct that will have have methods or anything like that.

----
I think doing so keeps the pattern and usage of these `<...>Options` structs very simple.